### DISCUSSION TITLE
inFilter with large array doesn't work

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#26837] [Questions] [answered]
Hello everyone,
I'm reaching out for some assistance before opening an issue, as I'm having trouble pinpointing the problem.

I have a join table called offers_merchants, and I need to query this table to get merchants by filtering with offer_id. I have a list of offerIDs (approximately 650 items) that I obtain from another query (essentially the same query, involving a join and an in filter). I then use this list to filter the offers_merchants table.

Here is the code snippet:
```
final rawOffers = await supabase
    .from('offers_cards')
    .select('offer(*)')
    .inFilter('card_id', selectedCardsIDs);
final offers = rawOffers.map((e) => Offer.fromJson(e['offer'])).toList();
final offersIDs = offers.map((e) => e.id).toList();

final rawMerchants = await supabase
    .from('offers_merchants')
    .select('*, merchant(*)')
    .inFilter('offer_id', offersIDs);

final merchants = rawMerchants.map((e) => Merchant.fromJson(e['merchant'])).toList();
```
When I run the above code, I receive a 400 Bad Request error. However, if I hardcode the list with 5-6 IDs and use it as a filter, the request works as intended.
Does anyone know why this might be happening? Is there something I'm missing?

Thank you for your help!

**Comment by GaryAustin1** [comment, marked as answer]
GET call URL's have a maximum length on most servers.  The filter info goes in the URL.
You will need use an rpc call which is a POST call and uses the body for the parameters.

**Reply by GaryAustin1** [reply]
https://github.com/supabase/postgrest-js/issues/393
